api/server/server.go

```diff
@@ -133,7 +136,18 @@ func (a *Server) RegisterAPIHandler(handler Handler, authBackend shttp.Authentic
 
 				resource := handler.New()
 
-				if err := common.JSONDecode(r.Body, &resource); err != nil {
+				var err error
+				if contentType := r.Header.Get("Content-Type"); contentType == "application/yaml" {
+					if content, e := ioutil.ReadAll(r.Body); e == nil {
+						err = yaml.Unmarshal(content, resource)
```

Comment:
For the `yaml.Unmarshal` to work, we need to add `yaml:MyAttribute` tags for all the API resources (right now, only the `Workflow` type have these tags)